### ISSUE TITLE
Poetry.lock updates failing

### ISSUE THREAD

**Issue by dominik-bln** [renovatebot/renovate#4843]
**What Renovate type are you using?**

Renovate self-hosted for self-hosted Gitlab

**Describe the bug**

Yesterday renovate stopped updating our `poetry.lock` file located in a subfolder `api/`. This has been working fine for months and we didn't change any configuration, just applied automatic updates to renovate.

**Did you see anything helpful in debug logs?**

```
 INFO: Failed to update api/poetry.lock file (repository=alpha/alpha, dependencies=pytest-mock, branch=ren/pytest-mock-1.x)
       "err": {
         "killed": false,
         "code": 1,
         "signal": null,
         "cmd": "poetry update --lock --no-interaction pytest-mock",
         "stdout": "Creating virtualenv alpha-api-py3.8 in /home/ubuntu/.cache/pypoetry/virtualenvs\nThe virtual environment was not created successfully because ensurepip is not\navailable.  On Debian/Ubuntu systems, you need to install the python3-venv\npackage using the following command.\n\n    apt-get install python3-venv\n\nYou may need to use sudo with that command.  After installing the python3-venv\npackage, recreate your virtual environment.\n\nFailing command: ['/home/ubuntu/.cache/pypoetry/virtualenvs/alpha-api-py3.8/bin/python', '-Im', 'ensurepip', '--upgrade', '--default-pip']\n\n",
         "stderr": "",
         "message": "Command failed: poetry update --lock --no-interaction pytest-mock\n",
         "stack": "Error: Command failed: poetry update --lock --no-interaction pytest-mock\n\n    at ChildProcess.exithandler (child_process.js:294:12)\n    at ChildProcess.emit (events.js:198:13)\n    at ChildProcess.EventEmitter.emit (domain.js:448:20)\n    at maybeClose (internal/child_process.js:982:16)\n    at Socket.stream.socket.on (internal/child_process.js:389:11)\n    at Socket.emit (events.js:198:13)\n    at Socket.EventEmitter.emit (domain.js:448:20)\n    at Pipe._handle.close (net.js:606:12)"
       }
 INFO: files committed (repository=alpha/alpha, dependencies=pytest-mock, branch=ren/pytest-mock-1.x)
 INFO: PR created (repository=alpha/alpha, dependencies=pytest-mock, branch=ren/pytest-mock-1.x)
       "pr": 3650
```

**Comment by JamieMagee** [comment]
This looks like it might be related to the recent Python 3.8 update.

Can you provide some more information about you're running Renovate? Is it via npm/npx or Docker?

**Comment by dominik-bln** [comment]
We are running it via the provided Docker image.

**Comment by rarkins** [comment]
As a non-Python user, I'm not sure yet if the answer is either:
1. Install `python3-venv` like it's saying, or
2. Fix up the config so that it's not using venv like before

**Comment by dominik-bln** [comment]
Hm, I'm also not sure, we are completely sidestepping venv for our Docker Python builds because unlike renovate we only need one environment.

Looks to me like this is more of an upstream issue with Poetry now though, i. e. maybe going with option 1 would be good enough and then hope that poetry releases a new version that fixes the problem?

**Comment by JamieMagee** [comment]
Yeah, I think this is already tracked in #4839. We're waiting for a poetry release of `0.12.18` or `1.0.0`.

**Comment by dominik-bln** [comment]
I just played around with this in the renovate Dockerfile and the weird part is, that locking works if I run the command twice:

![Screenshot 2019-11-22 at 13 14 45](https://user-images.githubusercontent.com/855119/69424926-1a583f80-0d2a-11ea-8e8f-53be5ea637f4.png)

**Comment by dominik-bln** [comment]
We got more confirmation, that only the first locking is problematic for `poetry`. We had a successful creation of a poetry.lock on a renovate run for `openpyxl`, but only after it first failed for `gunicorn`: 

```
INFO: Running poetry via global poetry (repository=alpha/alpha, dependencies=gunicorn, branch=ren/gunicorn-20.x)
 INFO: Failed to update api/poetry.lock file (repository=alpha/alpha, dependencies=gunicorn, branch=ren/gunicorn-20.x)
       "err": {
         "killed": false,
         "code": 1,
         "signal": null,
         "cmd": "poetry update --lock --no-interaction gunicorn",
         "stdout": "Creating virtualenv alpha-api-py3.8 in /home/ubuntu/.cache/pypoetry/virtualenvs\nThe virtual environment was not created successfully because ensurepip is not\navailable.  On Debian/Ubuntu systems, you need to install the python3-venv\npackage using the following command.\n\n    apt-get install python3-venv\n\nYou may need to use sudo with that command.  After installing the python3-venv\npackage, recreate your virtual environment.\n\nFailing command: ['/home/ubuntu/.cache/pypoetry/virtualenvs/alpha-api-py3.8/bin/python', '-Im', 'ensurepip', '--upgrade', '--default-pip']\n\n",
         "stderr": "",
         "message": "Command failed: poetry update --lock --no-interaction gunicorn\n",
         "stack": "Error: Command failed: poetry update --lock --no-interaction gunicorn\n\n    at ChildProcess.exithandler (child_process.js:294:12)\n    at ChildProcess.emit (events.js:198:13)\n    at ChildProcess.EventEmitter.emit (domain.js:448:20)\n    at maybeClose (internal/child_process.js:982:16)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)"
       }
 INFO: files committed (repository=alpha/alpha, dependencies=gunicorn, branch=ren/gunicorn-20.x)
 INFO: PR title changed (repository=alpha/alpha, dependencies=gunicorn, branch=ren/gunicorn-20.x)
       "branchName": "ren/gunicorn-20.x",
       "oldPrTitle": "fix(deps): update dependency gunicorn to v20.0.2",
       "newPrTitle": "fix(deps): update dependency gunicorn to v20.0.3"
 INFO: PR updated (repository=alpha/alpha, dependencies=gunicorn, branch=ren/gunicorn-20.x)
       "committedFiles": true,
       "pr": 3670
 INFO: Branch needs creating (repository=alpha/alpha, dependencies=openpyxl, branch=ren/openpyxl-3.x)
 INFO: Running poetry via global poetry (repository=alpha/alpha, dependencies=openpyxl, branch=ren/openpyxl-3.x)
 INFO: Updated lockfile (repository=alpha/alpha, dependencies=openpyxl, branch=ren/openpyxl-3.x)
       "seconds": 43,
       "type": "api/poetry.lock",
       "stdout": "Updating dependencies\nResolving dependencies...\n\nWriting lock file\n\n\n",
       "stderr": "PyPI: An error occurred when reading setup.py or setup.cfg: 'summary'\n"
 INFO: files committed (repository=alpha/alpha, dependencies=openpyxl, branch=ren/openpyxl-3.x)
 INFO: PR created (repository=alpha/alpha, dependencies=openpyxl, branch=ren/openpyxl-3.x)
       "pr": 3675
 INFO: Created Merge Request #3675 (repository=alpha/alpha, dependencies=openpyxl, branch=ren/openpyxl-3.x)
```

**Comment by rarkins** [comment]
@dominik-bln that's a really strange observation!
